torch/csrc/jit/passes/utils/memory_dag.h

```diff
@@ -97,7 +97,11 @@ struct Element {
   MemoryLocations pointedFrom;
 
   // Elements can contain other elements (e.g. List[Tensor])
-  MemoryLocations containedElements;
+  // This is a vector instead of bitset because index matters.
+  //
+  // NOTE: elements MAY BE nullptr; this represents that they are non-aliasing
+  // types.
+  std::vector<Element*> containedElements;
```

Comment:
If we really want to maintain the indexing, we need a set per index. 
otherwise with something like:
```
if cond: 
    y = (tensor, tensor)
else:
    y = (tensor, tensor)
```

the output of the if node needs a different set per index. as is, we're just appending, and it has a vector of length 4